NEWS.md

```diff
@@ -1,60 +1,55 @@
+   `add-contextual-data()` block. (#2936)
+ * Send Server Name Identification (SNI) information with `transport(tls)`.
+   Enable it by setting the `sli(yes)` option in the `tls` block in your
+   `destination`. (#2930)
 
- * Add relocation support for `disk-buffer()`. The `relocate` subcommand of
-   `dqtool` can be used to move a single or multiple queue files. (#2855)
-
- * `file(), pipe()`: The `time-reap()` option now can be set or disabled for
-   each destination separately.
-   Use `time-reap(0)` to disable closing idle destination files. (#2798)
+## Features
 
- * `syslog-ng-ctl`: Add `list-files` subcommand to print files present in the
-   current configuration. (#2797)
+ * Timezone related `rewrite` rules (#2818)
+ * `templates`: change the `$LOGHOST` macro to honour `use-fqdn()` (#2894)
+ * `syslog-ng-sysconfdir` define (#2932)
+ * Dqtool assign to persist (#2872)
 
 ## Bugfixes
 
- * Fix minor memory leaks (#2868)
- * Add global context to standalone parsers (#2876)
- * Fix heap usage after free in cfg_run_parser_with_main_context (#2884)
- * Fix g_thread_init call order issue with older glibs (#2853)
- * SNMP destination: fixing statistics format (#2854)
- * eventlog: flush escaped_buffer when full (#2837)
- * dbparser: remove unnecessary lock (#2838)
- * dbparser: fix crash when context times out in the middle of another rule (#2832)
- * radix: fix grouping in PCRE (#2808)
- * add-contextual-data: make filters config plugin aware (#2886)
- * Undefined warning regression (#2829)
+ * Fix backtick substitution (#2906, #2909)
+ * Fix SCL block parameter substitution  (#2901)
+ * `logthrsource/fetcher`: crash after failed reload (#2907)
+ * Fix crash at shutdown on 32bit systems (#2893, #2895)
+ * Invalidate `LEGACY_MSGHDR` when `PID` or `PROGRAM` is `unset` (#2896)
+ * on 32bit platform `diskq` ftruncate could fail due to size 32/64 interface (#2892)
+ * Fix zoneinfo parser (#2898)
+ * `afinet`: fix multicast ip detection (#2905)
+ * Fix hostname resolve on systems with only the loopback network interface configured (#2933)
+ * Fix `wildcard-file()` option validation (#2922)
+ * `kafka-c`: Fix multiple memleaks (#2944)
 
 ## Other changes
 
- * Disable `time-reap()` on non-templated filenames by default (#2798)
- * The `--preprocess-into` command line flag accepts `-`, and writes the
-   preprocessed configuration to stdout. (#2869)
- * Add information on environment variables passed to the confgen script (#2888)
- * Light: minor fixes (#2867, #2844)
- * python: use malloc_debug for python unit tests (#2866)
- * Travis: verbose unit test output (#2851)
- * Travis cleanup (#2809)
- * filter: add unit tests to `filter-op` (#2835)
- * Fix clang/gcc diagnostic differences (#2810)
- * Fix leak in stats test (#2874)
- * cmake, autotools: -Wundef to enable-extra-warning  (#2806)
+ * `geoip`: remove deprecated module, `geoip2` database location detection (#2780)
+ * `persist-state`: abort when mapping invalid handle (#2945)
```

Comment:
We should either omit this or tell when the abort was possible.

Reply:
I omitted it.